schema_based_on_diagram.sql

```diff
@@ -0,0 +1,34 @@
+CREATE DATABASE clinic;
+CREATE TABLE patients(
+  id             INT GENERATED ALWAYS AS IDENTITY,
+  name            VARCHAR(120),
+  date_of_birth   DATE,
+  PRIMARY KEY(id)
+);
+
+
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT REFERENCES patients(id),
+    status VARCHAR(80),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE treatments (
+	id INT GENERATED ALWAYS AS IDENTITY,
+  type	VARCHAR(120), 
+  name	VARCHAR(120),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories_treatments(
+    medical_history_id INTEGER REFERENCES medical_histories(id),
+    treatment_id INTEGER REFERENCES treatments(id),
+    PRIMARY KEY (medical_history_id, treatment_id)
+);
+
+CREATE TABLE invoices { id INT GENERATED ALWAYS AS IDENTITY, total_amount DECIMAL, generated_at TIMESTAMP, payed_at TIMESTAMP, medical_history_id INT REFERENCES medical_histories(id), 

```

Comment:
* Kindly ensure that all FOREIGN KEYS have indexes created [As stated in the requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes.)